Explain arrow operator in class chapter

km45 opened the pull request.

km45 committed:
Explain arrow operator in class chapter

cpmrph commented on docs/ch07-01-member-variables.md:
````diff
@@ -22,3 +22,40 @@ Rectangle r;
+クラスのオブジェクトをポインタで扱う場合に、メンバ変数を参照するには次のようにします。
+
+```cpp
+std::unique_ptr<Rectangle> r(new Rectangle);
````
![badge](https://img.shields.io/badge/review-imo-orange.svg)
この例の使い方的には問題にはならないですが、
`Rectangle` のようなPOD型は `new Rectangle()` として
値初期化したほうがベターだと思いました。

https://wandbox.org/permlink/63EvIixaIpvUbHGb
https://stackoverflow.com/questions/620137/do-the-parentheses-after-the-type-name-make-a-difference-with-new

km45 replied:
この時点ではコンストラクタを説明していないため、
`Rectangle()` という表記の意味が分からないのではと思ってこうしました。

`new` しないでポインタを使用すれば避けられそうなため、以下のように修正します。

```cpp
Rectangle r1;
Rectangle* r2 = &r1;
(*r2).height_ = 10;
```

km45 replied:
修正しました。
6f46052

km45 committed:
Fix sample code

Two expressions `new Rectangle` and `new Rectangle()` are different [^1] for POD.
We should use `new Rectangle()` as example code, but default contsructor has not been explained at this page.
So change example code not to use `new` (explicitly dynamic memory allocation).

[^1]: https://stackoverflow.com/questions/620137/do-the-parentheses-after-the-type-name-make-a-difference-with-new
